### ISSUE TITLE
`fs::read_link` on OS X removes the root, returning an invalid path.

### ISSUE THREAD

**Issue by jakerr** [rust-lang/rust#28095]
The [documents](http://doc.rust-lang.org/nightly/std/fs/fn.read_link.html) are not very clear about what it should return but I have to guess it should be an absolute path:
``
Reads a symbolic link, returning the file that the link points to.
``
It should say "(absolute) path to the file" or something.

Reproduction:
* notice on OS X  `/var` is a symlink to `/private/var`

```rust
use std::fs;
use std::path;
fn main(){
    let p = path::PathBuf::from("/var");
    println!("{:?}", fs::read_link(&p));
}
```

Outputs:
```
Ok("private/var")
```

Notice the leading `/` is missing.

**Comment by jakerr** [comment]
Apparently the readlink command on OS X has the same behavior making me less confident that this is a bug.

```
 > readlink /var
private/var
```

What I don't understand is when it prepends the `/` and when it doesn't. If I readlink my ~/.vim I get an absolute path to the directory :confused: 
```
 > readlink ~/.vim
/Users/jakerr/dotfiles/dot/vim
```

**Comment by jakerr** [comment]
I see, the `/var` symlink in OS X is a relative symlink which is why it is reported like so. To handle the output of `read_link` correctly you need to check if it is a relative path and handle accordingly.

Looks like **working as intended**
